esp8266/machine_rtc.c

```diff
@@ -55,23 +56,96 @@ uint64_t pyb_rtc_alarm0_expiry; // in microseconds
 
 // RTC overflow checking
 STATIC uint32_t rtc_last_ticks;
+STATIC uint32_t rtc_last_cal;
+// Clock overflow checking
+STATIC uint32_t clk_last_ticks;
+STATIC uint64_t clk_offset;
+
+void esp_clk_set_us_since_2000(uint64_t nowus) {
+    // Get the current clock tick
+    clk_last_ticks = system_get_time();
+    // Set current time as base for future calculations
+    clk_offset = nowus;
+};
+
+uint64_t esp_clk_get_us_since_2000() {
+    uint64_t offset;
+    uint32_t clk_ticks;
+
+    offset = clk_offset;
+    clk_ticks = system_get_time();
+
+    if (clk_ticks >= clk_last_ticks) {
+      offset+= clk_ticks-clk_last_ticks;
+    } else {
+      // If overflow happened, assume 1 wrap-around and persist info for the new cycle
+      offset+= clk_ticks+~clk_last_ticks+1;
```

Comment:
I saw that one. However, I think that variable is for slightly different purposes.
The "system clock" is designed to be adjustable, i.e. if for any reason a program wants to adjust current system time, the "offset" will be amended accordingly;
The system_time_high_word is for tracking the time differences between two ref points, and adjusting system clock should not affect this counter, otherwise the result will be wrong after adjusting the system clock.